export/index.template.html

```diff
@@ -671,6 +689,111 @@ <h2>Message log</h2>
+            height:
+              "^(\\d+(px|em|rem|%|vh|vw|in|cm|mm|pt|pc|ex|ch|vmin|vmax)|auto)$",
+            maxWidth:
+              "^(\\d+(px|em|rem|%|vh|vw|in|cm|mm|pt|pc|ex|ch|vmin|vmax)|none)$",
+            maxHeight:
+              "^(\\d+(px|em|rem|%|vh|vw|in|cm|mm|pt|pc|ex|ch|vmin|vmax)|none)$",
+            lineHeight:
+              "^(\\d+(\\.\\d+)?(px|em|rem|%|vh|vw|in|cm|mm|pt|pc|ex|ch|vmin|vmax)|normal)$",
+            boxShadow:
+              "^(none|(\\d+(px|em|rem) ?){2,4} (#[0-9a-f]{3,8}|rgba?\\(\\d{1,3}, \\d{1,3}, \\d{1,3}(, \\d?(\\.\\d{1,2})?)?\\)) ?(inset)?)$",
+            textAlign: "^(left|right|center|justify|initial|inherit)$",
+            overflowWrap: "^(normal|break-word|anywhere)$",
+            wordWrap: "^(normal|break-word)$",
+            resize: "^(none|both|horizontal|vertical|block|inline)$",
+          };
+
+          Object.entries(styles).forEach(([property, value]) => {
+            const styleProperty = property.trim();
+            if (styleProperty.length === 0) {
+              throw new Error("css style property cannot be empty");
+            }
+            const styleRegexStr = cssValidationRegex[styleProperty];
+            if (!styleRegexStr) {
+              throw new Error(
+                `invalid or unsupported css style property: "${styleProperty}"`
+              );
+            }
+            const styleRegex = new RegExp(styleRegexStr);
+            const styleValue = value.trim();
+            if (styleValue.length == 0) {
+              throw new Error(`css style for "${styleProperty}" is empty`);
+            }
+            const isValidStyle = styleRegex.test(styleValue);
+            if (!isValidStyle) {
+              throw new Error(
+                `invalid css style value for property "${styleProperty}"`
+              );
+            }
+            validStyles[styleProperty] = styleValue;
+          });
+
+          return validStyles;
+        }
+
+        /**
+         * Function to apply settings on this page. For now, the only settings that can be applied
+         * are for "styles". Upon successful application, return the valid, sanitized settings JSON string.
+         * @param {string} settings
+         * @return {string}
+         */
+         function applySettings(settings) {
+          const validSettings = {};
+          if (!settings) {
+            return JSON.stringify(validSettings);
+          }          
+          const settingsObj = JSON.parse(settings);
+          if (settingsObj.styles) {
+            // Valid styles will be applied the "key-div" textarea HTML element.
+            const keyDivTextarea = document.getElementById("key-div");
+            if (!keyDivTextarea) {
+              throw new Error(
+                "no key-div HTML element found to apply settings to."
+              );
+            }
+
+            // Validate, sanitize, and apply the styles to the "key-div" textarea.
```

Comment:
The `key-div` element is just a simple div, which later gets its `innerText` set

Reply:
changed this